specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2023-04-01-preview/dataprotection.json

```diff
@@ -0,0 +1,7956 @@
+        "vaultCriticalOperation": {
+          "type": "string"
+        },
+        "defaultResourceRequest": {
+          "type": "string"
+        }
+      }
+    },
+    "ResourceGuardProxyBase": {
+      "description": "ResourceGuardProxyBase object, used in ResourceGuardProxyBaseResource",
+      "type": "object",
+      "properties": {
+        "resourceGuardResourceId": {
+          "type": "string"
+        },
+        "resourceGuardOperationDetails": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceGuardOperationDetail"
+          },
+          "x-ms-identifiers": []
+        },
+        "lastUpdatedTime": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        }
+      }
+    },
+    "ResourceGuardProxyBaseResource": {
+      "description": "ResourceGuardProxyBaseResource object, used for response and request bodies for ResourceGuardProxy APIs",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DppResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ResourceGuardProxyBase",
+          "description": "ResourceGuardProxyBaseResource properties"
+        }
+      }
+    },
+    "ResourceGuardProxyBaseResourceList": {
+      "description": "List of ResourceGuardProxyBase resources",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DppResourceList"
+        }
+      ],
+      "properties": {
+        "value": {
+          "description": "List of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceGuardProxyBaseResource"
+          }
+        }
+      }
+    },
+    "UnlockDeleteRequest": {
+      "description": "Request body of unlock delete API.",
+      "type": "object",
+      "properties": {
+        "resourceGuardOperationRequests": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "resourceToBeDeleted": {
+          "type": "string"
+        }
+      }
+    },
+    "UnlockDeleteResponse": {
+      "description": "Response of Unlock Delete API.",
+      "type": "object",
+      "properties": {
+        "unlockDeleteExpiryTime": {
+          "description": "This is the time when unlock delete privileges will get expired.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "VaultName": {
+      "name": "vaultName",
+      "in": "path",
+      "description": "The name of the backup vault.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "BackupInstanceName": {
+      "name": "backupInstanceName",
+      "in": "path",
+      "description": "The name of the backup instance.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ResourceId": {
```

Comment:
Should add [x-ms-skip-url-encoding](https://azure.github.io/autorest/extensions/#x-ms-skip-url-encoding) extension to skip url encoding for ResourceId

Reply:
The ResourceId is not known to be url encoded, so I don't believe we need to have this attribute to ResourceId field

Reply:
It should be, otherwise the input ResourceId will be encoded and not able to present the original value.

Reply:
![image](https://user-images.githubusercontent.com/5196139/236753684-b5fe1174-dc78-40fd-bd23-b3111273e689.png)

Reply:
Okay, done

Reply:
Done